### DISCUSSION TITLE
Data source suggestion: SQLite

### DISCUSSION THREAD

**Discussion by everythingability** [Budibase/budibase#1486] [Ideas]
I wondered if this might both be easy to implement and/or really useful?

SQLite is so ubiquitous and is often in my case used at the beginning of an idea. Being able to use it in Budibase would be cool. Either direct access, but really I'd prefer it was more of an import into Budibase's format, like with csv.

Thanks

Tom

**Comment by mjashanks** [comment]
Thanks for the suggestion Tom!

SQLite might be a bit weird as a direct connection, as the self-hosted server would need access to the data file. Would probably have to put it on a shared drive or something.

However, I agree that it would be a good one as an initial import. 

To be honest, I can't see us doing this soon. However, different import sources is a topic that comes up very often - so it's definitely one that we should consider at some stage. Ideally, we would want to enable community developed import sources - so that anyone can come and add a new import type.

Thanks!

**Reply by OpenGG** [reply]
@mjashanks 

Are there any factors that a contributor should take into consideration when attempting to improve support for Sqlite?

I have been thinking about adding support for Sqlite as a data source by starting with copying and modifying the integration code from postgres.

However, these two databases have distinct capabilities, and I'm concerned that the limitations of Sqlite, being less feature-rich compared to postgres, could result in incomplete or dysfunctional functionality.

**Reply by melohagan** [reply]
Hey @OpenGG The safest bet may be to build a [custom datasource plugin](https://docs.budibase.com/docs/custom-datasource)

**Comment by christophberger** [comment]
Several distributed SQLite solutions have emerged recently (LiteFS, dqlite, rqlite), as well as a SQLite server (libsql with their [sqld](https://github.com/libsql/sqld-dev) project), so it might be worth re-visiting this idea. 

The distributed SQLite variants would remove the need for a shared drive to have other apps access the data file.

The `sqld` server even adds SQLite to the range of server-based relational databases.

What do you think, @mjashanks?

**Reply by fredguth** [reply]
I would add 'Turso' to this llist as well!
https://turso.tech

**Reply by christophberger** [reply]
...which is basically a hosted service on top of libsql/sqld. Yes, it would be great to be able to connect to Turso DBs.

**Comment by emmayu** [comment]
hello, I am looking for sqlite for Budibase data source, do we have any more updates on this topic yet? Thank you!
